pkg/raft/rawnode.go

```diff
@@ -298,51 +297,19 @@ func newStorageAppendRespMsg(r *raft, rd Ready) pb.Message {
 		//     that no in-progress appends might overwrite them before removing entries
 		//     from the unstable log.
 		//
-		// To prevent these kinds of problems, we also attach the current term to the
-		// MsgStorageAppendResp (above). If the term has changed by the time the
-		// MsgStorageAppendResp if returned, the response is ignored and the unstable
-		// log is not truncated. The unstable log is only truncated when the term has
-		// remained unchanged from the time that the MsgStorageAppend was sent to the
-		// time that the MsgStorageAppendResp is received, indicating that no-one else
-		// is in the process of truncating the stable log.
+		// If accTerm has changed by the time the MsgStorageAppendResp is returned,
+		// the response is ignored and the unstable log is not truncated. The
+		// unstable log is only truncated when the term has remained unchanged from
+		// the time that the MsgStorageAppend was sent to the time that the response
+		// is received, indicating that no new leader has overwritten the log.
 		//
-		// However, this replaces a correctness problem with a liveness problem. If we
```

Comment:
We would still have this liveness problem if `accTerm` could be bumped without having new entries. But today, `accTerm` is either bumped on a snapshot (when we clear the `unstable` log), or on appending a [non-empty](https://github.com/cockroachdb/cockroach/blob/9bfd93727b23b4ff38fba468d5aa3bb5e289b9e3/pkg/raft/log.go#L151-L159) slice of `entries`. In the former case, we don't need the last accepted term logic because there are no pending unstable entries. In the latter case, we send messages because `len(Entries) > 0`.